### ISSUE TITLE
Request for Kuramoto dataset

### ISSUE THREAD

**Issue by hilbert9221** [ethanfetaya/NRI#23]
According to Section 5.1 of the original paper, I use the code by Laszuk (https://github.com/laszukdawid/Dynamical-systems/blob/master/kuramoto.py) to simulate the Kuramoto model. The settings are listed as follows.

> N = 5	# number of particles
> intrinsic frequencies \omega uniformly sampled from [1, 10)
> initial phases \phi uniformly sampled from [0, 2\pi)
> coupling constants k_{ij} = 1 with probability 0.5
> subsample factor  = 10
> length of trajectories T = 50
> particle states x = (d\phi / dt, sin \phi, \omega)

For normalization, I use the function `load_kuramoto_data` from `utils.py`.

Some important settings of NRI are listed as follows.
> encoder: CNN
> decoder: MLP
> skip_first = True
> lr = 5e-4
> prediction_step = 10	# teacher forcing in every 10-th time step

It seems I've strictly followed the settings of the original paper, but the accuracy gets stucked at around 54%, and the mse gets stucked at the level of 1e-1.  There must be some mistakes in simulation or training. Do you have any advice? Would you mind providing a copy of Kuramoto dataset to help me out?

**Comment by tkipf** [comment]
You can download our code for generating the Kuramoto dataset used in our
paper here: https://www.dropbox.com/s/3jwim0n7iijrz9w/kuramoto.zip?dl=0

**Comment by hilbert9221** [comment]
Thank you! I'll try your code.

**Comment by hilbert9221** [comment]
Your code does help. But I haven't got comparable results when the number of objects is 10. Do you have any advice for data generation or model training to promote the performance? By the way, here are the brief results.
| method | acc (N=5) | acc (N=10) |
| :----: | :----: | :----: |
| reported | __96.0%__      | __75.7%__         |
| reproduced | 94.6% | 67.6% |

**Comment by tkipf** [comment]
Have you tried setting `skip_first` to True?

**Comment by hilbert9221** [comment]
Yeah, I set 'skip_first=True' as it makes a difference.  I also set 'encoder=CNN'. Other settings are the same as described in the paper, and most of them are the default settings in the code.

I wonder if I should try more training data. Maybe 50k is not be enough for 10-object system.
